qiskit/circuit/parameterexpression.py

```diff
@@ -37,6 +37,7 @@ def __init__(self, symbol_map, expr):
         """
         self._parameter_symbols = symbol_map
         self._symbol_expr = expr
+        self._lambda = None
```

Comment:
The test failures look to be due to the function object returned by `lambdify` not being able to be pickled.